### PR TITLE
WIP chore: Stream wrappers

### PR DESCRIPTION
**Pull request by dranikpg**
Will help with #3775

### CODE REVIEW

File: src/server/stream_family.cc [hunk 1]
```diff
@@ -603,6 +603,55 @@ int StreamTrim(const AddTrimOpts& opts, stream* s) {
...
+
+  T* operator->() {
+    return ptr_;
+  }
+
+ protected:
+  T* ptr_;
+};
+
+struct GroupWrapper : public PointerWrapper<streamCG> {
```
Comment (romange) [inline src/server/stream_family.cc:626]: call it StreamPtr

File: src/server/stream_family.cc [hunk 2]
```diff
@@ -603,6 +603,55 @@ int StreamTrim(const AddTrimOpts& opts, stream* s) {
...
+  streamConsumer* FindConsumer(string_view name, sds& tmp_str, bool create = false) {
+    tmp_str = sdscpylen(tmp_str, name.data(), name.length());
+    auto* consumer = streamLookupConsumer(ptr_, tmp_str, SLC_NO_REFRESH);
+    if (consumer == nullptr && create)
+      return streamCreateConsumer(ptr_, tmp_str, NULL, 0, SCC_NO_NOTIFY | SCC_NO_DIRTIFY);
+    return consumer;
+  }
+};
+
+struct StreamWrapper : public PointerWrapper<stream> {
```
Comment (romange) [inline src/server/stream_family.cc:639]: call it StreamPtr

File: src/server/stream_family.cc [hunk 3]
```diff
@@ -603,6 +603,55 @@ int StreamTrim(const AddTrimOpts& opts, stream* s) {
...
+  explicit StreamWrapper(stream* s) : PointerWrapper{s} {
+  }
+  explicit StreamWrapper(const CompactObj& co) : StreamWrapper{static_cast<stream*>(co.RObjPtr())} {
+  }
+
+  static void Create(CompactObj* obj) {
```
Comment (romange) [inline src/server/stream_family.cc:645]: I do not see how it's related to this class, can be just another helper function

**Comment by dranikpg** [comment]
Not worth the 30 saved loc

**Pull request closed without merging**